nomad/leader.go

```diff
@@ -85,7 +87,7 @@ func (s *Server) monitorLeadership() {
 	wasLeader := false
 	for {
 		select {
-		case isLeader := <-s.leaderCh:
+		case isLeader := <-leaderCh:
```

Comment:
@i0rek This is new leadership function pattern.  Note the `leaderStep` transitions and removal of `s.leaderCh`.